### PR TITLE
Add clusterdeployment resource and ClusterDeployment type

### PR DESCRIPTION
**Pull request by zgalor**
Remove cluster delete force parameter

### CODE REVIEW

File: model/clusters_mgmt/v1/cluster_deployment_type.model [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Representation of a clusterdeployment.
+class ClusterDeployment {
+	// Content of the clusterdeployment.
+	Content []Interface
```
Comment (vkareh) [inline model/clusters_mgmt/v1/cluster_deployment_type.model:20]: Won't this result in an array of `content` items with a single element?
```
{
  "kind": "ClusterDeployment",
  "href": "/api/clusters_mgmt/v1/clusters/1k1eiapjh11fpgtqig80ihj83rq93e0g/clusterdeployment",
  "id": "clusterdeployment",
  "content": [
    {
      "kind": "ClusterDeployment",
      "apiVersion": "hive.openshift.io/v1",
      ...
    }
  ]
}
```

**Comment by zgalor** [comment]
@tzvatot @jhernand @igoihman @vkareh PTAL

File: model/clusters_mgmt/v1/cluster_deployment_type.model (hunk 1 above)
Comment (zgalor) [reply model/clusters_mgmt/v1/cluster_deployment_type.model:20]: Whoops :facepalm:

Comment (zgalor) [reply model/clusters_mgmt/v1/cluster_deployment_type.model]: fixed. Nice catch

File: model/clusters_mgmt/v1/cluster_deployment_type.model [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/cluster_deployment_type.model:2]: You know...

File: model/clusters_mgmt/v1/clusterdeployment_resource.model [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/clusterdeployment_resource.model:2]: same

**Comment by tzvatot** [comment]
Aside from the nitpick, LGTM

File: model/clusters_mgmt/v1/cluster_deployment_type.model (hunk 2 above)
Comment (zgalor) [reply model/clusters_mgmt/v1/cluster_deployment_type.model]: fixed

File: model/clusters_mgmt/v1/clusterdeployment_resource.model (hunk 3 above)
Comment (zgalor) [reply model/clusters_mgmt/v1/clusterdeployment_resource.model]: fixed

**Pull request merged**